### ISSUE TITLE
[Filtering] Fast Follow Fixes + UI Polish

### ISSUE THREAD

**Issue by nicklepine** [Shopify/dawn#100]
## Purpose

In order to ensure the feature was merged on time, we merged Faceted filters with some known issues.  The idea now is to prioritize and tackle them one-by-one to try and get as many fixed as possible in the meanwhile and all of them fixed before the merchant launch

### Visual

* [ ] The animation doesn't feel super smooth when the filtering sheet is invoked and then closed. / * Animation is broken when opening/closing the drawer
* [ ]  The checkboxes don't 100% match the style in the Figma file (I'm not sure where else in the theme we encounter checkboxes (customer pages?), but we should make sure the pattern is consistent throughout. The hover style is to be adjusted, and the the label could have its opacity increased to 100%. 

### a11y

* [x] On tablet and mobile, the pills should wear a different smaller style , and comply with the accessible tap target
* [ ] Tabbing can send you behind the drawer
* [x] Can scroll the elements behind the open drawer.  This scrolling should be blocked.

### Browser specific (may not be fixable):

* [ ] On my iPhone, the price filter still allows non numeric values.
* [ ] On my phone, with disabled JS, tapping sorting triggers a layout shift for a slit second.

### Missing feature

* [ ]  Sorting doesn't sync between desktop and mobile
* [ ]  The range slider on the price range

### Refactor

* [ ]  Make a custom element out of the active filters / clear all / reset button to avoid the need to re-bind events to them

**Comment by Oliviammarcello** [comment]
### WIP UI polish review:

- [ ] Align dropdown arrows to the same line as the text. [Screenshot](https://screenshot.click/21-34-6osvv-lwq9t.png)
- [ ] Move checkboxes slightly to the left to appear more aligned on both desktop and mobile. [Desktop screenshot](https://screenshot.click/21-35-a1j4z-xo8i2.png) | [Mobile screenshot](https://screenshot.click/21-38-eut2b-quey5.png)
- [ ] On mobile the "Clear" and "Apply" buttons should fix to the bottom of the screen. [Screenshot](https://screenshot.click/21-38-ak92r-kbz3h.png)
- [ ] The price filtering should have the "$" inside the input field. [Screenshot](https://screenshot.click/21-40-c3xty-3m152.png) | [Figma](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=7155%3A1783)
- [ ] On mobile there is a bit too much padding above the "Filter & sort" button" and more padding is required above the "Showing x products" text. [Screenshot](https://screenshot.click/21-58-g3kl8-vopzr.png)
- [ ] The loading icon should be aligned to the left on mobile. [Screenshot](https://screenshot.click/21-02-ith8x-caz3j.png)
- [ ]

**Comment by ludoboludo** [comment]
on iOS I'm seeing this for the checkboxes within the drawer: ![](https://screenshot.click/21-07-5qahc-vb9jj.png)

**Comment by tyleralsbury** [comment]
## Other small issues

* [ ] The mobile show filters button has uneven padding on the focus

**Comment by tyleralsbury** [comment]
> The checkboxes don't 100% match the style in the Figma file (I'm not sure where else in the theme we encounter checkboxes (customer pages?), but we should make sure the pattern is consistent throughout. The hover style is to be adjusted,

I didn't see anything for hover styles in the Figma.  @Oliviammarcello do you have any context?

**Comment by Oliviammarcello** [comment]
> > The checkboxes don't 100% match the style in the Figma file (I'm not sure where else in the theme we encounter checkboxes (customer pages?), but we should make sure the pattern is consistent throughout. The hover style is to be adjusted,
> 
> I didn't see anything for hover styles in the Figma. @Oliviammarcello do you have any context?

@tyleralsbury good point! Can we follow [this prototype?](https://screenshot.click/21-03-25tkb-82zza.mp4)
<img width="829" alt="Screenshot 2021-07-08 at 12 45 49 PM" src="https://user-images.githubusercontent.com/60230008/124962902-296a2f00-dfed-11eb-8ea4-a3b89e1ddc48.png">

**Comment by Oliviammarcello** [comment]
> > The checkboxes don't 100% match the style in the Figma file (I'm not sure where else in the theme we encounter checkboxes (customer pages?), but we should make sure the pattern is consistent throughout. The hover style is to be adjusted,
> 
> I didn't see anything for hover styles in the Figma. @Oliviammarcello do you have any context?

Good point! Let's follow [this prototype](https://screenshot.click/21-03-25tkb-82zza.mp4) for the hover styles
<img width="829" alt="Screenshot 2021-07-08 at 12 45 49 PM" src="https://user-images.githubusercontent.com/60230008/124977252-43f8d400-dffe-11eb-8522-416adf4199f2.png">

**Comment by tyleralsbury** [comment]
Update after discussion w/ @Oliviammarcello we won't be adding hover styles since checkboxes are pretty self-explanatory as being an interactive element.

**Comment by nicklepine** [comment]
New bugs found this week (there may be some overlap with what was already reported) 

**Describe the current behavior**

- [ ] Animation doesn't have smooth open transition. It blinks. It should feel smoother.


- [ ] The Filter and Sort button disappears from the page when the sheet is open. It should remain visible in the area where we still see the page. 
![image](https://user-images.githubusercontent.com/8127145/126511573-a4196a08-91d2-41ef-9a9d-51cc10c7bd18.png)


- [ ] When going into a filtering option, there is an extra left border around the drawer. This shouldn't happen. 
![image](https://user-images.githubusercontent.com/8127145/126511655-d2e71b1f-0233-48e5-80e8-53f4da04757c.png)

- [ ] When going into a filtering option, the filter heading is using the wrong font.
https://screenshot.click/21-03-whje0-a8zp3.png

- [ ] When going into a filtering option, the filter heading should align with the check boxes. 
https://screenshot.click/21-03-whje0-a8zp3.png

- [ ] Stroke-weight of Checkboxes should use the same strokeweight as other icons on the theme. 
https://screenshot.click/21-03-85vs0-frdrd.png

- [ ] Mis-aligned clear and apply buttons on iPhone
https://screenshot.click/21-03-85vs0-frdrd.png

- [ ] When going into a filtering option, the right spacing too tight
https://screenshot.click/21-04-wvmpa-mj5a4.png

- [ ] When going into the price filtering option, the clear and apply buttons don't display
https://screenshot.click/21-04-cfu41-ounhz.png

- [ ] When playing with filters on mobile view in editor, A shadow lingers on the “clear” button after it’s been selected. We should remove this entirely.
![image](https://user-images.githubusercontent.com/8127145/126512848-739645d6-0190-4c46-bae4-2a70bea878f0.png)

- [ ] On the sort, the sort option heading (i.e. Date: new to old) should be right aligned to the caret (rather than left aligned). 
https://screenshot.click/21-11-l9sxj-grwvr.png

**Comment by Oliviammarcello** [comment]
UI update to address new feedback:
- [ ] On desktop, "X products" will now remain inline with the filter and sort button when the filters a collapsed. [Figma.](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=11759%3A110099) This will remain the same position when pills are added. [Figma](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=11759%3A109880)
- [ ] When filters aren't collapsed, "X products" will be to the right of "sort by". [Figma](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=11759%3A109414)
- [ ] On a tablet, the space between sort by and the filters should change from 80px to 60px. [Figma.](https://www.figma.com/file/JcYdswrk2JhqBTGBVWPM7J/Dawn-2.0?node-id=11759%3A112255)

Nothing is changing on mobile. Thanks!

**Comment by KaichenWang** [comment]
@Oliviammarcello 

Summary of issues remaining:

### Cannot reproduce and need more information

- On my phone, with disabled JS, tapping sorting triggers a layout shift for a slit second.
- Animation doesn't have smooth open transition. It blinks. It should feel smoother.

### Should be addressed separately, because affects theme globally

- When scrolling the filter goes on top of the sticky nav. [Issue created](https://github.com/Shopify/dawn/issues/357)
- Tabbing can send you behind the drawer [Issue created](https://github.com/Shopify/dawn/issues/360)

### Out of scope for "design polish" - these will require additional r&d to implement. Separate issues should be created for each of these and their priorities should be assessed individually

- The range slider on the price range
- The price filtering should have the "$" inside the input field.
- Stroke-weight of Checkboxes should use the same strokeweight as other icons on the theme.
- On the sort, the sort option heading (i.e. Date: new to old) should be right aligned to the caret (rather than left aligned).
- Reset button stays active when there are no current active filters in a group. Would be nice to disable it so that the items don't refresh when it's clicked and nothing has changed.

### Focus rings - these should be addressed in conjunction with other potential changes to focus rings for consistency

- Can you add additional padding to the left and right side of the sort by active state.
- When playing with filters on mobile view in editor, A shadow lingers on the “clear” button after it’s been selected. We should remove this entirely.
- The mobile show filters button has uneven padding on the focus
